runtime/spring/src/main/java/com/flit/runtime/spring/FlitExceptionHandler.java

```diff
@@ -11,54 +15,67 @@
 import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
 
-import javax.servlet.http.HttpServletRequest;
-import java.util.HashMap;
-import java.util.Map;
-
 @ControllerAdvice
 @Component
 public class FlitExceptionHandler extends ResponseEntityExceptionHandler {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(FlitExceptionHandler.class);
+  private static final Logger LOGGER = LoggerFactory.getLogger(FlitExceptionHandler.class);
 
 
-    @ExceptionHandler(Exception.class)
-    public ResponseEntity<?> handleException(HttpServletRequest request, Exception e) {
-        LOGGER.error(
-            "Exception: request = {}, method = {}, msg = {}",
-            request.getRequestURI(), request.getMethod(), e.getMessage(), e
-        );
+  @ExceptionHandler(Exception.class)
+  public ResponseEntity<?> handleException(HttpServletRequest request, Exception e) {
+    LOGGER.error(
+        "Exception: request = {}, method = {}, msg = {}",
+        request.getRequestURI(), request.getMethod(), e.getMessage(), e
+    );
 
-        Map<String, Object> response = new HashMap<>();
-        response.put("code", ErrorCode.INTERNAL);
-        response.put("msg", "An internal error has occurred");
+    Map<String, Object> response = new HashMap<>();
+    response.put("code", ErrorCode.INTERNAL);
+    response.put("msg", "An internal error has occurred");
 
+    return ResponseEntity
+        .status(ErrorCode.INTERNAL.getHttpStatus())
+        .contentType(MediaType.APPLICATION_JSON)
+        .body(response);
+  }
 
-        return ResponseEntity
-            .status(ErrorCode.INTERNAL.getHttpStatus())
-            .contentType(MediaType.APPLICATION_JSON)
-            .body(response);
-    }
+  @ExceptionHandler(InvalidProtocolBufferException.class)
+  public ResponseEntity<?> handleInvalidProtocolBufferException(HttpServletRequest request,
+      InvalidProtocolBufferException e) {
+
+    LOGGER.error("InvalidProtocolBufferException: request = {}, method = {}, msg= {}",
+        request.getRequestURI(), request.getMethod(), e.getMessage(), e);
 
-    @ExceptionHandler(FlitException.class)
-    public ResponseEntity<?> handleFlitException(HttpServletRequest request, FlitException e) {
+    Map<String, Object> response = new HashMap<>();
+    response.put("code", ErrorCode.INVALID_ARGUMENT);
+    response.put("msg", "Invalid JSON");
```

Comment:
I think we'll want to preserve the error message because it's pretty helpful for debugging malformed requests (e.g. "Cannot find field: queues in message aqueduct.api.v1.SendRequest"), especially since `application/protobuf` or `application/json` can both throw `InvalidProtocolBufferException`, so "Invalid JSON" might not be accurate.

Reply:
sounds good!